server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/model/KeywordsTest.scala

```diff
@@ -0,0 +1,127 @@
+    assertThat(LENIENT_KEYWORDS_FACTORY.fromSet(Set(Keyword.ANSWERED)).asFlags).isEqualTo(new Flags(Flag.ANSWERED))
+  }
+
+  "asFlagsWithRecentAndDeletedFromShouldBuildFlagsFromKeywordsAndRecentOriginFlags" in {
+    val originFlags = FlagsBuilder.builder.add(Flag.RECENT, Flag.DRAFT).build
+    val expectedFlags = FlagsBuilder.builder.add(Flag.ANSWERED, Flag.RECENT).build
+    assertThat(LENIENT_KEYWORDS_FACTORY.fromSet(Set(Keyword.ANSWERED)).asFlagsWithRecentAndDeletedFrom(originFlags)).isEqualTo(expectedFlags)
+  }
+
+  "asFlagsWithRecentAndDeletedFromShouldBuildFlagsFromKeywordsWithDeletedAndRecentOriginFlags" in {
+    val originFlags = FlagsBuilder.builder.add(Flag.RECENT, Flag.DELETED, Flag.DRAFT).build
+    val expectedFlags = FlagsBuilder.builder.add(Flag.ANSWERED, Flag.RECENT, Flag.DELETED).build
+    assertThat(LENIENT_KEYWORDS_FACTORY.fromSet(Set(Keyword.ANSWERED)).asFlagsWithRecentAndDeletedFrom(originFlags)).isEqualTo(expectedFlags)
+  }
+
+  "asMapShouldReturnEmptyWhenEmptyMapOfStringAndBoolean" in {
+    assertThat(LENIENT_KEYWORDS_FACTORY.fromSet(Set.empty).keywords.nonEmpty).isFalse
+  }
+
+  "asMapShouldReturnMapOfStringAndBoolean" in {
+    val expectedMap = Map(Keyword.of("$Answered") -> Keyword.FLAG_VALUE)
+    assertThat(LENIENT_KEYWORDS_FACTORY.fromSet(Set(Keyword.ANSWERED)).asMap).isEqualTo(expectedMap)
+  }
+
+  "throwWhenUnsupportedKeywordShouldThrowWhenHaveUnsupportedKeywords" in {
+    assertThatThrownBy(() => STRICT_KEYWORDS_FACTORY.fromSet(Set(Keyword.DRAFT, Keyword.DELETED))).isInstanceOf(classOf[IllegalArgumentException])
+  }
+
+  "throwWhenUnsupportedKeywordShouldNotThrowWhenHaveDraft" in {
+    val keywords = STRICT_KEYWORDS_FACTORY.fromSet(Set(Keyword.ANSWERED, Keyword.DRAFT))
+    assertThat(keywords.getKeywords).isEqualTo(Set(Keyword.ANSWERED, Keyword.DRAFT))
+  }
+
+  "filterUnsupportedShouldFilter" in {
+    val keywords = LENIENT_KEYWORDS_FACTORY.fromSet(Set(Keyword.ANSWERED, Keyword.DELETED, Keyword.RECENT, Keyword.DRAFT))
+    assertThat(keywords.getKeywords).isEqualTo(Set(Keyword.ANSWERED, Keyword.DRAFT))
+  }
+
+  "containsShouldReturnTrueWhenKeywordsContainKeyword" in {
+    val keywords = LENIENT_KEYWORDS_FACTORY.fromSet(Set(Keyword.SEEN))
+    assertThat(keywords.contains(Keyword.SEEN)).isTrue
+  }
+
+  "containsShouldReturnFalseWhenKeywordsDoNotContainKeyword" in {
+    val keywords = LENIENT_KEYWORDS_FACTORY.fromSet(Set())
+    assertThat(keywords.contains(Keyword.SEEN)).isFalse
+  }
+
+  "fromListShouldReturnKeywordsFromListOfStrings" in {
+    val keywords = LENIENT_KEYWORDS_FACTORY.fromCollection(Set("$Answered", "$Flagged"))
+    assertThat(keywords.getKeywords).isEqualTo(Set(Keyword.ANSWERED, Keyword.FLAGGED))
+  }
+
+  "fromListShouldNotThrowOnInvalidKeywordForLenientFactory" in {
+    assertThat(LENIENT_KEYWORDS_FACTORY.fromCollection(Set("in&valid")).keywords).isEqualTo(Keywords.DEFAULT_VALUE)
+  }
+
+  "fromMapShouldNotThrowOnInvalidKeywordForLenientFactory" in {
```

Comment:
Replace camel case with spaces to enhance readability.

I propose: `fromMap should not throw on invalid keyword for lenientFactory`

Reply:
Yeah, very agree with you, i will change the name of the others too